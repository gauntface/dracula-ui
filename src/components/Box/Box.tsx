import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import { spacingClasses, SpacingPropType } from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full',
}

/**
 * Box Props
 */
export interface BoxProps<K extends keyof Element = 'div'>
  extends AllHTMLAttributes<K> {
  /** The background color. */
  color?: keyof typeof colors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** Dracula UI standard spacing properties. */
  spacing?: SpacingPropType

  /** The HTML element to be used */
  as?: K
}

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.color && colors[props.color],
      props.rounded && roundedBorders[props.rounded],
      ...spacingClasses(props.spacing),
      props.className
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
