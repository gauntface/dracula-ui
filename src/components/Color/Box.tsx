import React, { HTMLAttributes } from 'react'

export const colors = {
  black: 'drac-bg-black',
  white: 'drac-bg-white',
  cyan: 'drac-bg-cyan',
  green: 'drac-bg-green',
  orange: 'drac-bg-orange',
  pink: 'drac-bg-pink',
  purple: 'drac-bg-purple',
  red: 'drac-bg-red',
  yellow: 'drac-bg-yellow',
  purpleCyan: 'drac-bg-purple-cyan',
  yellowPink: 'drac-bg-yellow-pink',
  cyanGreen: 'drac-bg-cyan-green',
  pinkPurple: 'drac-bg-pink-purple',
  animated: 'drac-bg-animated'
}

export interface BoxProps extends HTMLAttributes<HTMLDivElement> {
  color?: keyof typeof colors
}

export const Box: React.FC<BoxProps> = props => {
  const finalProps = {
    ...props,
    className: `drac-box ${colors[props.color ?? 'black']}`
  }

  return <div {...finalProps} />
}
