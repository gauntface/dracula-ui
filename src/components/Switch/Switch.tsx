import { ColorMap } from '../../base/colors'
import React, { HTMLAttributes } from 'react'
import {
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'
import classNamesDedupe from 'classnames/dedupe'

export const switchColors: Partial<ColorMap> = {
  white: 'drac-switch-white',
  cyan: 'drac-switch-cyan',
  green: 'drac-switch-green',
  orange: 'drac-switch-orange',
  pink: 'drac-switch-pink',
  purple: 'drac-switch-purple',
  red: 'drac-switch-red',
  yellow: 'drac-switch-yellow'
}

/** Switch Props. */
export interface SwitchProps
  extends HTMLAttributes<HTMLInputElement>,
    PaddingMixin,
    MarginMixin {
  /** The name of the form element. */
  name?: string

  /**
   * The Dracula UI theme color to be applied to the Switch.
   */
  color: keyof typeof switchColors

  disabled?: boolean
}

/**
 * Switch is an abstraction pattern that represents toggles.
 *
 * Use the Switch component as an alternative for Checkboxes or Radios
 * when options are binary.
 *
 * Switches are implemented as Checkboxes under the hood,
 * and work normally with forms.
 */
export const Switch: React.FC<SwitchProps> = (props: SwitchProps) => {
  const finalProps = {
    ...props,
    className: classNamesDedupe(
      `drac-switch`,
      'drac-checkbox',
      props.className,
      switchColors[props.color],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  return <input type="checkbox" {...finalProps} />
}

Switch.displayName = 'Switch'
