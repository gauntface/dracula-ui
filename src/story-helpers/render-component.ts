import testRenderer from 'react-test-renderer'

export type ComponentExample = {
  code: ReturnType<testRenderer.ReactTestRenderer['toJSON']>
  docs: string
  title: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: React.ReactElement,
  docs: string = ''
): ComponentExample {
  const rendered = testRenderer.create(comp)
  const code = rendered.toJSON()
  return {
    title,
    code,
    docs
  }
}
