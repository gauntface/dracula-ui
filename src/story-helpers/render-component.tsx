import React from 'react'
import Enzyme, { mount,  shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import prettier from 'prettier'
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: React.ReactElement,
  docs: string = ''
): ComponentExample {

  
  const html = mount(comp).html()
  const react = toJson(shallow(<div>{comp}</div>))

  const prettyHTML = prettier.format(html, {
    parser: 'html'
  })

  return {
    title,
    html: prettyHTML,
    react,
    docs
  }
}
