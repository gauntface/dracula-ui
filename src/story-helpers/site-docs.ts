import {
  renderSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import React from 'react'

// import * as serializer from 'jest-snapshot-serializer-raw'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  describe(`Site: ${componentClass.displayName ?? componentClass.name}`, () => {
    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp(),
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
    })

    const variations = documentation.variations()

    variations.forEach(variation => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp(),
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()
      })
    })
  })
}
