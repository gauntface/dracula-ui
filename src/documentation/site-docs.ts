import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '../story-helpers/render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from '../story-helpers/pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name

  describe(`Site: ${name}`, () => {
    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)
      const dspPath = `${process.cwd()}/dsp/data/components/${name}.json`
      await fs.writeFile(dspPath, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
