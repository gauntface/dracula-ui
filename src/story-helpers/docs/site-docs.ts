import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import React from 'react'
import fs from 'fs'
import { pretty } from '@/story-helpers/pretty'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name

  describe(`Site: ${name}`, () => {
    let examples: Record<string, ComponentExample> = {}

    afterAll(() => {
      const path = `${process.cwd()}/examples/${name}.json`
      fs.writeFileSync(path, pretty(JSON.stringify(examples), 'json'))
    })

    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
