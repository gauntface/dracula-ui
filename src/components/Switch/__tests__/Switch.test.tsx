import React from 'react'
import { Switch } from '@/components/Switch/Switch'
import {
  SwitchColors,
  SwitchDisabled,
  SwitchNormal
} from '@/components/Switch/__stories__/Switch.stories'
import { docs } from '@/documentation/site-docs'
import { snapshot } from '../../../documentation/render-component'
import { Box } from '../../Box/Box'

docs(Switch, {
  basic() {
    return snapshot(
      'Usage',
      () => (
        <Box>
          <Switch color="purple" id="demo" name="demo" defaultChecked />
          <label htmlFor="demo" className="drac-text drac-text-white">
            Checked
          </label>
        </Box>
      ),
      'Render a toggle component using checkbox under the hood.'
    )
  },
  variations() {
    return [
      snapshot(
        'Disabled',
        () => (
          <Box>
            <Box>
              <Switch
                id="one"
                disabled
                defaultChecked
                color="white"
                name="one"
              />
              <label htmlFor="one" className="drac-text drac-text-white">
                Checked
              </label>
            </Box>
            <Box>
              <Switch
                id="two"
                disabled
                color="white"
                name="two"
              />
              <label htmlFor="two" className="drac-text drac-text-white">
                Unchecked
              </label>
            </Box>
          </Box>
        ),
        'A disabled switch is unusable and un-clickable.'
      ),
      snapshot(
        'Colors',
        () => (
          <Box>
            <Switch color="cyan" id="colors" name="colors" />
            <label htmlFor="colors" className="drac-text drac-text-cyan">
              Checked
            </label>
          </Box>
        ),
        'Switches can be customized to use any of the Dracula UI theme colors.'
      )
    ]
  }
})
