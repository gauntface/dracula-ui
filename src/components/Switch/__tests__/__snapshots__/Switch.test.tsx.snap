// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Switch Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column; padding: 3rem\\">
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"one\\"
      color=\\"purple\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-purple\\"
      checked=\\"\\"
    /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Option 1</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"two\\"
      color=\\"purple\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-purple\\"
    /><label for=\\"two\\" class=\\"drac-text drac-text-white\\">Option 2</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"three\\"
      color=\\"purple\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-purple\\"
    /><label for=\\"three\\" class=\\"drac-text drac-text-white\\">Option 3</label>
  </div>
</div>
",
  "react": "<div style={{...}}>
  <div style={{...}}>
    <Switch id=\\"one\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
      Option 1
    </label>
  </div>
  <div style={{...}}>
    <Switch id=\\"two\\" color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"two\\" className=\\"drac-text drac-text-white\\">
      Option 2
    </label>
  </div>
  <div style={{...}}>
    <Switch id=\\"three\\" color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"three\\" className=\\"drac-text drac-text-white\\">
      Option 3
    </label>
  </div>
</div>",
  "title": "Usage",
}
`;

exports[`Site: Switch Colors 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column; padding: 3rem\\">
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-black</p>
    <input
      type=\\"checkbox\\"
      id=\\"black\\"
      color=\\"black\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-black\\"
      checked=\\"\\"
    /><label for=\\"black\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-white</p>
    <input
      type=\\"checkbox\\"
      id=\\"white\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-white\\"
      checked=\\"\\"
    /><label for=\\"white\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-cyan</p>
    <input
      type=\\"checkbox\\"
      id=\\"cyan\\"
      color=\\"cyan\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-cyan\\"
      checked=\\"\\"
    /><label for=\\"cyan\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-green</p>
    <input
      type=\\"checkbox\\"
      id=\\"green\\"
      color=\\"green\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-green\\"
      checked=\\"\\"
    /><label for=\\"green\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-orange</p>
    <input
      type=\\"checkbox\\"
      id=\\"orange\\"
      color=\\"orange\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-orange\\"
      checked=\\"\\"
    /><label for=\\"orange\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-pink</p>
    <input
      type=\\"checkbox\\"
      id=\\"pink\\"
      color=\\"pink\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-pink\\"
      checked=\\"\\"
    /><label for=\\"pink\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-purple</p>
    <input
      type=\\"checkbox\\"
      id=\\"purple\\"
      color=\\"purple\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-purple\\"
      checked=\\"\\"
    /><label for=\\"purple\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-red</p>
    <input
      type=\\"checkbox\\"
      id=\\"red\\"
      color=\\"red\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-red\\"
      checked=\\"\\"
    /><label for=\\"red\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-yellow</p>
    <input
      type=\\"checkbox\\"
      id=\\"yellow\\"
      color=\\"yellow\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-yellow\\"
      checked=\\"\\"
    /><label for=\\"yellow\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-animated</p>
    <input
      type=\\"checkbox\\"
      id=\\"animated\\"
      color=\\"animated\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox\\"
      checked=\\"\\"
    /><label for=\\"animated\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">.drac-switch .drac-switch-cyanGreen</p>
    <input
      type=\\"checkbox\\"
      id=\\"cyanGreen\\"
      color=\\"cyanGreen\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox\\"
      checked=\\"\\"
    /><label for=\\"cyanGreen\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-pinkPurple
    </p>
    <input
      type=\\"checkbox\\"
      id=\\"pinkPurple\\"
      color=\\"pinkPurple\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox\\"
      checked=\\"\\"
    /><label for=\\"pinkPurple\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-purpleCyan
    </p>
    <input
      type=\\"checkbox\\"
      id=\\"purpleCyan\\"
      color=\\"purpleCyan\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox\\"
      checked=\\"\\"
    /><label for=\\"purpleCyan\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
  <div>
    <p class=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-yellowPink
    </p>
    <input
      type=\\"checkbox\\"
      id=\\"yellowPink\\"
      color=\\"yellowPink\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox\\"
      checked=\\"\\"
    /><label for=\\"yellowPink\\" class=\\"drac-text drac-text-white\\">Switch</label>
  </div>
</div>
",
  "react": "<div style={{...}}>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      black
    </p>
    <Switch id=\\"black\\" defaultChecked={true} color=\\"black\\" name=\\"demo\\" />
    <label htmlFor=\\"black\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      white
    </p>
    <Switch id=\\"white\\" defaultChecked={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"white\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      cyan
    </p>
    <Switch id=\\"cyan\\" defaultChecked={true} color=\\"cyan\\" name=\\"demo\\" />
    <label htmlFor=\\"cyan\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      green
    </p>
    <Switch id=\\"green\\" defaultChecked={true} color=\\"green\\" name=\\"demo\\" />
    <label htmlFor=\\"green\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      orange
    </p>
    <Switch id=\\"orange\\" defaultChecked={true} color=\\"orange\\" name=\\"demo\\" />
    <label htmlFor=\\"orange\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      pink
    </p>
    <Switch id=\\"pink\\" defaultChecked={true} color=\\"pink\\" name=\\"demo\\" />
    <label htmlFor=\\"pink\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      purple
    </p>
    <Switch id=\\"purple\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"purple\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      red
    </p>
    <Switch id=\\"red\\" defaultChecked={true} color=\\"red\\" name=\\"demo\\" />
    <label htmlFor=\\"red\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      yellow
    </p>
    <Switch id=\\"yellow\\" defaultChecked={true} color=\\"yellow\\" name=\\"demo\\" />
    <label htmlFor=\\"yellow\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      animated
    </p>
    <Switch id=\\"animated\\" defaultChecked={true} color=\\"animated\\" name=\\"demo\\" />
    <label htmlFor=\\"animated\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      cyanGreen
    </p>
    <Switch id=\\"cyanGreen\\" defaultChecked={true} color=\\"cyanGreen\\" name=\\"demo\\" />
    <label htmlFor=\\"cyanGreen\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      pinkPurple
    </p>
    <Switch id=\\"pinkPurple\\" defaultChecked={true} color=\\"pinkPurple\\" name=\\"demo\\" />
    <label htmlFor=\\"pinkPurple\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      purpleCyan
    </p>
    <Switch id=\\"purpleCyan\\" defaultChecked={true} color=\\"purpleCyan\\" name=\\"demo\\" />
    <label htmlFor=\\"purpleCyan\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
  <div>
    <p className=\\"drac-text drac-text-white\\">
      .drac-switch .drac-switch-
      yellowPink
    </p>
    <Switch id=\\"yellowPink\\" defaultChecked={true} color=\\"yellowPink\\" name=\\"demo\\" />
    <label htmlFor=\\"yellowPink\\" className=\\"drac-text drac-text-white\\">
      Switch
    </label>
  </div>
</div>",
  "title": "Colors",
}
`;

exports[`Site: Switch Disabled 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column; padding: 3rem\\">
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"one\\"
      disabled=\\"\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-white\\"
      checked=\\"\\"
    /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Checked</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      id=\\"two\\"
      disabled=\\"\\"
      color=\\"white\\"
      name=\\"demo\\"
      class=\\"drac-switch drac-checkbox drac-switch-white\\"
    /><label for=\\"two\\" class=\\"drac-text drac-text-white\\">Unchecked</label>
  </div>
</div>
",
  "react": "<div style={{...}}>
  <div style={{...}}>
    <Switch id=\\"one\\" disabled={true} defaultChecked={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
      Checked
    </label>
  </div>
  <div style={{...}}>
    <Switch id=\\"two\\" disabled={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"two\\" className=\\"drac-text drac-text-white\\">
      Unchecked
    </label>
  </div>
</div>",
  "title": "Disabled",
}
`;
